jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/milestones/CaseMilestoneListPresenter.java

```diff
@@ -48,19 +67,35 @@ protected void clearCaseInstance() {
 
     @Override
     protected void loadCaseInstance(final CaseInstanceSummary cis) {
-        refreshData(caseId);
+        refreshData();
     }
 
     protected void searchCaseMilestones() {
-        refreshData(caseId);
+        refreshData();
+    }
+
+    protected void refreshData() {
+        visibleCaseMilestones.clear();
```

Comment:
if i remove where these get cleared, the results keep compounding (since they are actually managed in the ViewImpl and not the presenter)